### PR TITLE
Hide add to your OC at the public page when sending shares is not all…

### PR DESCRIPTION
**Pull request by VicDeo**
…owed by admin

## Description
When ` Allow users on this server to send shares to other servers` is unchecked at the Settings->Admin-Sharing there should be no 'Add to your ownCloud' button
Bonus: switch to EventDispatcher as https://github.com/owncloud/core/pull/31944 allows it

## Related Issue
- Fixes https://github.com/owncloud/enterprise/issues/3291

## How Has This Been Tested?
1. Create a public link
2. Tick Settings->Admin-Sharing ` Allow users on this server to send shares to other servers`
3.  open the public link in another browser
Expected:  'Add to your ownCloud' at the public page

1. Create a public link
2. Untick Settings->Admin-Sharing ` Allow users on this server to send shares to other servers`
3.  open the public link in another browser
Expected:  There is NO 'Add to your ownCloud' at the public page

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: changelog/unreleased/37232 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Bugfix: Hide add to your OC at the public page when it's not allowed
+
+'Add to your ownCloud' button was removed from the public link page if ougoing federated sharing is disabled.
```
[the remark this answers is not in the document]
Comment (VicDeo) [reply changelog/unreleased/37232]: Thanks, updated.

**Comment by phil-davis** [comment]
I created issue #37264 to create some acceptance tests for this button existing or not.
If those get done quickly (tomorrow morning) then they can be added to this PR, otherwise if this PR gets merged first then we will add them as a separate PR.

**Review by phil-davis — COMMENTED** [review]
LGTM and works - from a QA/testing POV the acceptance tests demostrate that.
Someone else can approve the detail of the code changes.

**Review by jvillafanez — APPROVED** [review]

**Comment by phil-davis** [comment]
@VicDeo there were acceptance tests here, and now they have disappeared.

**Comment by VicDeo** [comment]
@phil-davis oops...I didn't pull before rebasing :(

**Review by phil-davis — APPROVED** [review]
All the code is there now.

**Comment by VicDeo** [comment]
can't merge - codecov stalled :(

**Comment by phil-davis** [comment]
`codecov/patch` result has been not coming many times in recent days. I will submit a result...

**Pull request merged**